File: benchmarks/string/regex_100k.py
import time, re
lines = ["The quick brown fox jumps over the lazy dog #{}".format(i) for i in range(100000)]
pattern = re.compile(r'\b\w{5,}\b')
start = time.monotonic_ns()
count = sum(len(pattern.findall(line)) for line in lines)
end = time.monotonic_ns()
print((end - start) // 1000)

File: benchmarks/string/regex_100k.rb
lines = Array.new(100000) { |i| "The quick brown fox jumps over the lazy dog ##{i}" }
pattern = /\b\w{5,}\b/
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
count = lines.sum { |line| line.scan(pattern).length }
finish = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
puts(finish - start)
